doc/explanation/components/reactive_html_components.md

```diff
@@ -0,0 +1,635 @@
+
+    _template = """
+    <label id="text-field" class="mdc-text-field mdc-text-field--filled">
+      <span class="mdc-text-field__ripple"></span>
+      <span class="mdc-floating-label">Label</span>
+      <input id="text-input" type="text" class="mdc-text-field__input" aria-labelledby="my-label" value="${value}"></input>
+      <span class="mdc-line-ripple"></span>
+    </label>
+    """
+
+    _dom_events = {'text-input': ['change']}
+
+    # By declaring an _extension_name the component should be loaded explicitly with pn.extension('material-components')
+    _extension_name = 'material-components'
+
+    _scripts = {
+        'render': "mdc.textField.MDCTextField.attachTo(text_field);"
+    }
+
+    __javascript__ = [
+        'https://unpkg.com/material-components-web@latest/dist/material-components-web.min.js'
+    ]
+
+    __css__ = [
+        'https://unpkg.com/material-components-web@latest/dist/material-components-web.min.css'
+    ]
+
+pn.extension('material-components') # for notebook
+
+text_field = MaterialTextField(value="Some value")
+
+pn.Column(text_field, text_field.param.value).servable()
+```
+
+In a notebook dependencies for this component will not be loaded unless the user explicitly loads them with a `pn.extension('material-components')`. In a server context you will also have to explicitly load this extension unless the component is rendered on initial page load, i.e. if the component is only added to the page in a callback you will also have to explicitly run `pn.extension('material-components')`.
+
+### Bootstrap
+
+[Bootstrap](https://getbootstrap.com/) is one of the most popular design frameworks. **We recommend not using [Bootstrap](https://getbootstrap.com/) with Panel**.
+
+You can use its CSS to style your components, but n our experience its javascript does not work well with Panel. It simply cannot select and update HTML elements inside the *shadowroot* of `ReactiveHTML` components.
+
+### Web Components
+
+[Web Components](https://en.wikipedia.org/wiki/Web_Components) are custom HTML elements. Web components work really great with `ReactiveHTML` and in the same way as built in HTML elements like `button`, `div` and `img`.
+
+Some web components that work well with Panels designs are
+
+- [Shoelace](https://shoelace.style/): A large, mature collection of components that integrate well with the Bootstrap design.
+- [Fast](https://explore.fast.design/): A relatively large and mature collection of components that integrate well with the Fast design.
+- [Material](https://github.com/material-components/material-web): A growing collection of components that integrate well with the Material design.
+
+For more inspiration check out the [awesome-web-components](https://github.com/web-padawan/awesome-web-components#components) list.
+
+### React, Preact and Vue
+
+`ReactiveHTML` can be used with [React](https://react.dev/), [Preact](https://preactjs.com/) and [Vue](https://vuejs.org/).
```

Comment:
I'm no longer sure `ReactiveHTML` works with React (and MaterialUI). In https://github.com/mui/material-ui/issues/17473 they claim React and Material does not and do not want to support ShadowRoot.

There is a minimal example here that I tried to get working but could not https://discourse.holoviz.org/t/help-with-panel-react-materialui/5204/3?u=marc